### PR TITLE
Work on developing the extension detail page and hooking it up with the list view

### PR DESCRIPTION
**Pull request by elvisisking**
- Work relating to #78 
- Initial implementation of the extension detail page
- See commit message for more details
- Forgot to mention in commit message that I added the Dashboard routes and resolvers files.

### CODE REVIEW

File: packages/api/src/WithExtension.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          actions: [],
+          extensionType: 'Steps',
+          name: '',
+          schemaVersion: '',
+        }}
+        initialValue={this.props.initialValue}
+      >
+        {({ read, response }) =>
+          this.props.children({
+            ...response,
```
Comment (riccardo-forina) [inline packages/api/src/WithExtension.tsx]: unless you have to change the response somehow, I think you can just pass the `response` object as is

File: packages/ui/src/Customization/ExtensionDetail.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+.extension-title {
+  margin: 0;
+}
+
+/* .section-heading {
+} */
+
+button + button {
```
Comment (riccardo-forina) [inline packages/ui/src/Customization/ExtensionDetail.css]: Namespace this please

File: packages/ui/src/Customization/ExtensionOverview.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.property-label {
```
Comment (riccardo-forina) [inline packages/ui/src/Customization/ExtensionOverview.css]: namespace should be the same as the filename

File: packages/api/src/WithExtension.tsx (hunk 1 above)
Comment (elvisisking) [reply packages/api/src/WithExtension.tsx]: I'll remove `read`.

File: packages/ui/src/Customization/ExtensionDetail.css (hunk 2 above)
Comment (elvisisking) [reply packages/ui/src/Customization/ExtensionDetail.css]: Will do

File: packages/ui/src/Customization/ExtensionOverview.css (hunk 3 above)
Comment (elvisisking) [reply packages/ui/src/Customization/ExtensionOverview.css]: Will do

**Pull request merged**